libs/net/services/Helpers/ApiService.cs

```diff
@@ -118,6 +118,41 @@ protected async Task<T> RetryRequestAsync<T>(Func<Task<T>> callbackDelegate)
             return await RetryRequestAsync<T>(callbackDelegate);
         }
     }
+
+    /// <summary>
+    /// Keep trying a request if the failure is caused by an optimistic concurrency error.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="callbackDelegate"></param>
+    /// <returns></returns>
+    public async Task<T> HandleConcurrencyAsync<T>(Func<Task<T>> callbackDelegate)
+    {
+        // Keep trying to update the record and handle concurrency errors.
+        while (true)
```

Comment:
Essentially keep trying until successful.  It will only error out if the exception is not a optimistic concurrency error.